browser/js/app-initializer.js

```diff
@@ -0,0 +1,170 @@
+import Loader from './loader';
+import ads from '../../components/n-ui/ads';
+import tracking from '../../components/n-ui/tracking';
+import date from 'o-date';
+import header from '../../components/n-ui/header';
+import oCookieMessage from 'o-cookie-message';
+import footer from '../../components/n-ui/footer';
+import { lazyLoad as lazyLoadImages } from 'n-image';
+import * as serviceWorker from 'n-service-worker';
+import DesktopAppBanner from 'n-desktop-app-banner';
+import * as syndication from 'n-syndication';
+import { perfMark, broadcast } from 'n-ui-foundations';
+import { waitForCondition } from './utils';
+
+export const presets = {
+	discrete: {
+		header: true,
+		footer: true,
+		date: true
+	},
+	complete: {
+		header: true,
+		footer: true,
+		date: true,
+		cookieMessage: true,
+		ads: true,
+		syndication: true
+	}
+};
+
+// Dispatch a custom `ftNextLoaded` event after the app executes.
+function dispatchLoadedEvent () {
+	let ftNextLoaded = false;
+	const ftNextLoadedTrigger = () => {
+		if (document.readyState === 'complete' && ftNextLoaded === false) {
+			ftNextLoaded = true;
+			window.dispatchEvent(new CustomEvent('ftNextLoaded'));
+			return true;
+		}
+	};
+	if (!ftNextLoadedTrigger()) {
+		window.addEventListener('load', ftNextLoadedTrigger);
+		document.onreadystatechange = ftNextLoadedTrigger;
+	}
+}
+
+export class AppInitializer {
+
+	initEnv () {
+
+		this.appInfo = {
+			isProduction: document.documentElement.hasAttribute('data-next-is-production'),
+			version: document.documentElement.getAttribute('data-next-version'),
+			name: document.documentElement.getAttribute('data-next-app'),
+			product: document.documentElement.getAttribute('data-next-product')
+		};
+
+		const flags = window.FT.flags.reduce((obj, flag) => {
```

Comment:
worth ensuring that `window.FT.flags` exists?

Reply:
I'll add a todo to tidy up definition of window.FT